### PR TITLE
[iOS/macOS] Rename onboarding shared pixel params and bucket daysSinceInstall

### PR DESCRIPTION
**Pull request by rachelmcr**
Task/Issue URL: https://app.asana.com/1/137249556945/project/1202818073656551/task/1217051716311541?focus=true
Tech Design URL: 
CC:

### Description

Updates the onboarding shared pixel parameters as agreed in [New parameter names and buckets](https://app.asana.com/1/137249556945/project/1212901916671993/task/1216821984327708?focus=true):

* Uses more transparent parameter names: `event`, `installType`, `daysSinceInstall`
* Buckets the `daysSinceInstall` values, no longer dropping values over 28 (instead using a `28+` bucket)

### Testing Steps

1. Build and run the macOS app.
2. Debug → Reset Data → Reset Onboarding
3. Close the window and reopen it.
4. Confirm `m_mac_onboarding_welcome` is shown with parameters `event`, `installType`, `daysSinceInstall` (no parameters `e`, `it`, `d`).
5. Confirm `daysSinceInstall` uses a bucket value corresponding to the days since the app was originally installed (e.g. `0`, `1-3`, `28+`).

### Impact

None: Internal tooling

---
###### Internal references:
[Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f) | [Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552) | [Tech Design Template](https://app.asana.com/0/59792373528535/184709971311943)

---

> [!NOTE]
> **Low Risk**
> Analytics-only pixel parameter changes with no user-facing behavior; downstream dashboards must expect new keys and bucket values instead of legacy `e`/`it`/`d`.
> 
> **Overview**
> Aligns **onboarding shared pixels** on iOS and macOS with clearer parameter names and bucketed install-age reporting.
> 
> **Parameter renames** on fired pixels: `e` → **`event`**, `it` → **`installType`**, `d` → **`daysSinceInstall`**. The handler and `OnboardingSharedPixelEvent` parameters map, plus iOS/macOS **`params_dictionary.json5`**, are updated together.
> 
> **`daysSinceInstall`** is no longer a raw day count capped at 28 (older installs omitted the param). It now sends string buckets: `0`, `1-3`, `4-10`, `11-28`, and **`28+`** for 29+ days. Tests cover each bucket and the negative-day omission case.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit f658b1285958db5993514f202c8fb10fb13cb178. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by rachelmcr** [comment]
Note: We shouldn't merge this until the pipeline is updated to accommodate the changes: https://app.asana.com/1/137249556945/project/1108686900785972/task/1216871249295897?focus=true

File: SharedPackages/Onboarding/Sources/OnboardingTests/OnboardingSharedPixelTests.swift [hunk 1]
```diff
@@ -165,72 +165,131 @@ final class OnboardingSharedPixelTests: XCTestCase {
...
-        XCTAssertEqual(event.additionalParameters?["d"], "28")
+        XCTAssertEqual(event.additionalParameters?["daysSinceInstall"], "0")
     }
 
-    func testWhenDaysSinceInstallIsOutOfRangeThenDParameterIsOmitted() throws {
-        let pixelFiring = PixelKitMock()
+    func testWhenDaysSinceInstallIsOneToThreeThenDaysSinceInstallBucketIsOneToThree() throws {
         let currentDate = Date()
-        let pixelHandler = makeHandler(installDateProvider: { currentDate.daysAgo(29) }, currentDateProvider: { currentDate }, pixelFiring: pixelFiring)
+        for daysAgo in [1, 3] {
```
Comment (alessandroboron) [inline SharedPackages/Onboarding/Sources/OnboardingTests/OnboardingSharedPixelTests.swift:219]: nip: should we get a random element in the test intervals? E.g. (1..3).randomElement()

**Review by alessandroboron — APPROVED** [review]
LGTM! Works as expected

File: SharedPackages/Onboarding/Sources/OnboardingTests/OnboardingSharedPixelTests.swift (hunk 1 above)
Comment (rachelmcr) [reply SharedPackages/Onboarding/Sources/OnboardingTests/OnboardingSharedPixelTests.swift:219]: I chose this approach because it tests both boundaries of each interval, which is more reliable than only testing a single random element. If it were a more expensive test I'd go with a random element but I don't think there's harm in testing the boundaries here.
